### DISCUSSION TITLE
Paperless-ngx services not running anymore, update doesn't work

### DISCUSSION THREAD

**Discussion by z3bbl** [tteck/Proxmox#3321] [Help Needed] [answered]
After some weeks my paperless instance running pretty good I noticed yesterday my paperless-ngx instance is not reachable via local ip address with port 8000 anymore. I checked the services and all services are failed:

`# systemctl status paperless-consumer paperless-webserver paperless-scheduler paperless-task-queue
x paperless-consumer.service - Paperless consumer
     Loaded: loaded (/etc/systemd/system/paperless-consumer.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-07-03 12:04:00 CEST; 19min ago
   Duration: 918ms
    Process: 428 ExecStart=python3 manage.py document_consumer (code=exited, status=1/FAILURE)
   Main PID: 428 (code=exited, status=1/FAILURE)
        CPU: 475ms

Jul 03 12:03:59 paperless-ngx python3[428]:   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
Jul 03 12:03:59 paperless-ngx python3[428]:   File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
Jul 03 12:03:59 paperless-ngx python3[428]:   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
Jul 03 12:03:59 paperless-ngx python3[428]:   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
Jul 03 12:03:59 paperless-ngx python3[428]:   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
Jul 03 12:03:59 paperless-ngx python3[428]:   File "/opt/paperless/src/documents/models.py", line 26, in <module>
Jul 03 12:03:59 paperless-ngx python3[428]:     from django_softdelete.models import SoftDeleteModel
Jul 03 12:03:59 paperless-ngx python3[428]: ModuleNotFoundError: No module named 'django_softdelete'
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-consumer.service: Main process exited, code=exited, status=1/FAILURE
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-consumer.service: Failed with result 'exit-code'.

x paperless-webserver.service - Paperless webserver
     Loaded: loaded (/etc/systemd/system/paperless-webserver.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-07-03 12:04:00 CEST; 19min ago
   Duration: 1.072s
    Process: 430 ExecStart=/usr/local/bin/gunicorn -c /opt/paperless/gunicorn.conf.py paperless.asgi:application (code=exited, status=1/FAILURE)
   Main PID: 430 (code=exited, status=1/FAILURE)
        CPU: 534ms

Jul 03 12:04:00 paperless-ngx gunicorn[430]:   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
Jul 03 12:04:00 paperless-ngx gunicorn[430]:   File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
Jul 03 12:04:00 paperless-ngx gunicorn[430]:   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
Jul 03 12:04:00 paperless-ngx gunicorn[430]:   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
Jul 03 12:04:00 paperless-ngx gunicorn[430]:   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
Jul 03 12:04:00 paperless-ngx gunicorn[430]:   File "/opt/paperless/src/documents/models.py", line 26, in <module>
Jul 03 12:04:00 paperless-ngx gunicorn[430]:     from django_softdelete.models import SoftDeleteModel
Jul 03 12:04:00 paperless-ngx gunicorn[430]: ModuleNotFoundError: No module named 'django_softdelete'
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-webserver.service: Main process exited, code=exited, status=1/FAILURE
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-webserver.service: Failed with result 'exit-code'.

x paperless-scheduler.service - Paperless Celery beat
     Loaded: loaded (/etc/systemd/system/paperless-scheduler.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-07-03 12:04:00 CEST; 19min ago
   Duration: 1.241s
    Process: 429 ExecStart=celery --app paperless beat --loglevel INFO (code=exited, status=1/FAILURE)
   Main PID: 429 (code=exited, status=1/FAILURE)
        CPU: 755ms

Jul 03 12:04:00 paperless-ngx celery[429]:   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
Jul 03 12:04:00 paperless-ngx celery[429]:   File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
Jul 03 12:04:00 paperless-ngx celery[429]:   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
Jul 03 12:04:00 paperless-ngx celery[429]:   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
Jul 03 12:04:00 paperless-ngx celery[429]:   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
Jul 03 12:04:00 paperless-ngx celery[429]:   File "/opt/paperless/src/documents/models.py", line 26, in <module>
Jul 03 12:04:00 paperless-ngx celery[429]:     from django_softdelete.models import SoftDeleteModel
Jul 03 12:04:00 paperless-ngx celery[429]: ModuleNotFoundError: No module named 'django_softdelete'
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-scheduler.service: Main process exited, code=exited, status=1/FAILURE
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-scheduler.service: Failed with result 'exit-code'.

x paperless-task-queue.service - Paperless Celery Workers
     Loaded: loaded (/etc/systemd/system/paperless-task-queue.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-07-03 12:04:00 CEST; 19min ago
   Duration: 1.299s
    Process: 431 ExecStart=celery --app paperless worker --loglevel INFO (code=exited, status=1/FAILURE)
   Main PID: 431 (code=exited, status=1/FAILURE)
        CPU: 752ms

Jul 03 12:04:00 paperless-ngx celery[431]:   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
Jul 03 12:04:00 paperless-ngx celery[431]:   File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
Jul 03 12:04:00 paperless-ngx celery[431]:   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
Jul 03 12:04:00 paperless-ngx celery[431]:   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
Jul 03 12:04:00 paperless-ngx celery[431]:   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
Jul 03 12:04:00 paperless-ngx celery[431]:   File "/opt/paperless/src/documents/models.py", line 26, in <module>
Jul 03 12:04:00 paperless-ngx celery[431]:     from django_softdelete.models import SoftDeleteModel
Jul 03 12:04:00 paperless-ngx celery[431]: ModuleNotFoundError: No module named 'django_softdelete'
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-task-queue.service: Main process exited, code=exited, status=1/FAILURE
Jul 03 12:04:00 paperless-ngx systemd[1]: paperless-task-queue.service: Failed with result 'exit-code'.`

When I tryed to update paperless-ngx via script I get following message:

✓ Stopped all Paperless-ngx Services
 | Updating to v2.10.2   
[ERROR] in line 81: exit code 0: while executing command pip install -r requirements.txt &> /dev/null

Current version is 2.8.5.

What can I do now?

**Comment by tteck** [comment, marked as answer]
https://github.com/tteck/Proxmox/discussions/3300#discussioncomment-9923938 then update again

**Reply by z3bbl** [reply]
Thank you very much. It works!
